### PR TITLE
Add cleanup functionality

### PR DESCRIPTION
**Pull request by florent-vial**
If the check box 'Clean first' is activated, repo forall 'git clean -fdx' will be executed first.

* added unit tests
* fixed typos in code

### CODE REVIEW

File: src/main/java/hudson/plugins/repo/RepoScm.java [hunk 1]
```diff
@@ -326,6 +334,7 @@ public RepoScm(final String manifestRepositoryUrl,
 				   final String repoUrl,
 				   final boolean currentBranch,
 				   final boolean resetFirst,
+				   final boolean cleanFirst,
```
Comment (rsandell) [inline src/main/java/hudson/plugins/repo/RepoScm.java]: This constructor is deprecated and only kept around for binary backward compatibility reasons, so the signature should not be changed.

Comment (florent-vial) [inline src/main/java/hudson/plugins/repo/RepoScm.java]: Done

**Comment by florent-vial** [comment]
Requested change has been made in https://github.com/jenkinsci/repo-plugin/pull/49/commits/9c7e81cbf0f0ff6414df31a5b782805c8725012a

**Comment by zwx168238** [comment]
i got a problem  my repo plugin versiovn was Version 1.10.7
but we did not find  check box 'Clean first' 
repo_version
repo_donwload

**Comment by zwx168238** [comment]
by get the release 1.10.7 source code , found missing function "clean first" .

**Comment by zwx168238** [comment]
@rsandell hi When will the plan release the next version?

**Pull request merged**